### PR TITLE
fix: Improved recurring event booking speed

### LINKED ISSUE #9517: [CAL-1947] Booking a recurring event type is slow
Found a bug? Please fill out the sections below. 👍

### Issue Summary

The availability loading is fast but It sometime takes \~10seconds or more to book the meeting. I have tried both on production and locally.

### Steps to Reproduce

1. Create a recurring event type
2. Make a booking

<sub>From [SyncLinear.com](https://synclinear.com) | [CAL-1947](https://linear.app/calcom/issue/CAL-1947/booking-a-recurring-event-type-is-slow)</sub>

### PR DESCRIPTION
**Pull request by JaideepGuntupalli**
## What does this PR do?

This pull request aims to improve the recurring booking speed by optimizing two crucial actions. The time required to perform the first action is directly proportional to the number of recurrences booked.

In the existing code, when 12 recurrences were booked, the first action took approximately 16 seconds. 

However, with the changes implemented in this pull request, the same 12 recurrences can be processed in just 9 seconds. 

This improvement is achieved by reducing the number of network requests from n to 1. Previously, the frontend code utilized a for loop, increasing network overhead. By streamlining the process, we significantly decrease network requests and payload size, enhancing the booking speed.

Although the network optimizations have been implemented, there is room for further improvement. One area that can be explored is reducing the number of exchanges from the database. It might be possible to create an optimized handler to enhance performance. Additionally, I couldn't find optimizations in the second action, which involves generating a booking success redirect. Further enhancements can be explored to make this action more efficient.

Fixes #9517 

## Type of change

- Chore (refactoring code, technical debt, workflow improvements)

## How should this be tested?

- Try to book a recurring event locally from the main branch code
- Try the same with the PR code

## Mandatory Tasks
- [x] Make sure you have self-reviewed the code. A decent size PR without self-review might be rejected.

### CODE REVIEW

File: apps/web/pages/api/book/event.ts [hunk 1]
```diff
@@ -8,6 +8,7 @@ async function handler(req: NextApiRequest & { userId?: number }, res: NextApiRe
   const session = await getServerSession({ req, res });
   /* To mimic API behavior and comply with types */
   req.userId = session?.user?.id || -1;
+  console.log("reqDAta: ", req.body);
```
Comment (alannnc) [inline apps/web/pages/api/book/event.ts:11]:
```suggestion
```

Comment (JaideepGuntupalli) [reply apps/web/pages/api/book/event.ts:11]: Yup, sorry i forgot lol.

**Comment by JaideepGuntupalli** [comment]
hey @emrysal, @hariombalhara, @zomars, just a reminder, can you pls review the PR

**Review by hariombalhara — APPROVED** [review]
That's an amazing improvement. Thank you so much !!

Excited to ship it !!
[excited](https://app.graphite.dev/macros/calcom)

**Pull request merged**